script/app.js

```diff
@@ -139,3 +147,77 @@ const completeTodo = (taskId) => {
     todos = updatedTodos;
     renderTodos();
 };
+
+const editTodo = (taskId) => {
+    const task = todos.find((task) => task.id === taskId);
+    editingId = task.id;
+
+    if (!task) {
+        alert("Task not found");
+        return;
+    }
+
+    const taskElement = document.getElementById(`task-${task.id}`);
+    const titleElement = taskElement.querySelector("h1");
+    const createdAtElement = taskElement.querySelector(".createdAt");
+    const inputElement = document.createElement("input");
+    inputElement.type = "text";
+    inputElement.value = task.title;
+    if (titleElement) {
+        titleElement.parentNode.replaceChild(inputElement, titleElement);
+    }
+    if (createdAtElement) {
+        createdAtElement.style.display = "none";
+    }
+    const editButton = taskElement.querySelector(".task-card__icon--edit");
+    editButton.innerText = "Save";
+
+    const saveHandler = () => {
+        const updatedTitle = sanitizeInput(inputElement.value.trim());
+
+        if (!updatedTitle) {
+            alert("Task title cannot be empty");
+            return;
+        }
+
+        const updatedTodos = [...todos];
+        const taskToUpdate = updatedTodos.find((task) => task.id === taskId);
+        taskToUpdate.title = updatedTitle;
+        todos = updatedTodos;
```

Comment:
why we need this?

Reply:
As before, we have done  completeTodo() with shallow copy. In this edit section, we also have updated task completion before saving when we are in edit mode. so that we must need the shallow copy. other wish it's not working as moc-up

Reply:
We are already creating a shallow copy at line 185